okdata/cli/commands/pubreg/questions.py

```diff
@@ -0,0 +1,200 @@
+        "type": "select",
+        "name": "env",
+        "message": "Maskinporten environment",
+        "choices": _environments,
+    }
+
+
+def q_team(team_client):
+    def _team_choices():
+        teams = team_client.get_teams(has_role="origo-team")
+
+        if not teams:
+            raise NoTeamError
+
+        return [Choice(t["name"], t["id"]) for t in teams]
+
+    return {
+        **_common_style,
+        "type": "select",
+        "name": "team_id",
+        "message": "Team",
+        "choices": _team_choices(),
+    }
+
+
+def q_provider():
+    return {
+        **_common_style,
+        "type": "select",
+        "name": "provider_id",
+        "message": "Provider",
+        "choices": [Choice(pname, pid) for pid, pname in _providers.items()],
+    }
+
+
+def q_scopes():
+    return {
+        **_common_style,
+        "type": "checkbox",
+        "name": "scopes",
+        "message": "Scopes",
+        "choices": lambda x: _scopes[x["provider_id"]],
+        "validate": (lambda choices: bool(choices) or "Select at least one scope"),
+    }
+
+
+def q_integration():
+    def _validate_integration(text):
+        if len(text) > 30:
+            return "Too long!"
+        if not re.fullmatch("[0-9a-z-]+", text):
+            return 'Only lowercase letters, numbers and "-", please'
+        return True
+
+    return {
+        **_common_style,
+        "type": "text",
+        "name": "integration",
+        "message": "Component/integration name",
+        "validate": _validate_integration,
+    }
+
+
+def q_client(pubreg_client):
+    def _client_choices(env):
+        clients = pubreg_client.get_clients(env)
+
+        if not clients:
+            raise NoClientsError
+
+        return [
+            Choice(
+                c["client_name"],
+                {"id": c["client_id"], "name": c["client_name"]},
+            )
+            for c in sorted(clients, key=itemgetter("client_name"))
+        ]
+
+    return {
+        **_common_style,
+        "type": "select",
+        "name": "client",
+        "message": "Client",
+        "choices": lambda x: _client_choices(x["env"]),
+    }
+
+
+def q_send_to_aws():
+    return {
+        **_common_style,
+        "type": "confirm",
+        "name": "send_to_aws",
+        "message": "Send key to AWS Parameter Store?",
+        "auto_enter": False,
+    }
+
+
+def q_aws_account():
+    return {
+        **_common_style,
+        "type": "text",
+        "name": "aws_account",
+        "message": "AWS account number",
+        "when": lambda x: x["send_to_aws"],
+        "validate": (
+            lambda t: bool(re.fullmatch("[0-9]{12}", t)) or "12 digits, please"
+        ),
+    }
+
+
+def q_aws_region():
+    return {
+        **_common_style,
+        "type": "select",
+        "name": "aws_region",
+        "message": "AWS region",
+        "when": lambda x: x["send_to_aws"],
+        "choices": [Choice(*r) for r in _aws_regions],
+    }
+
+
+def q_key(pubreg_client):
```

Comment:
Tror vi trenger en tilsvarende `NoKeysError`- ellers kaster questionary exception dersom det ikke finnes noen nøkler på klienten: `ValueError: A list of choices needs to be provided.`